### PR TITLE
Update Rule “graphql-when-to-use/rule”

### PR DESCRIPTION
**Pull request by ozairashfaqueSSW**
Automatically generated by Netlify CMS

### CODE REVIEW

File: rules/graphql-when-to-use/rule.md [hunk 1]
```diff
@@ -38,61 +36,53 @@ GraphQL differs from REST in that REST is concerned with  **resources** and Grap
...
 ### Advantages of GraphQL
 
-
 * **No under-fetching:** Sometimes a REST resource may only give you part of what you need. For example, if you need to know all the orders placed by a customer, what date they were ordered, and their current status, but the initial customer search resource only returns the details of the customer, this is called under-fetching. You then need to use the customer’s ID to query your orders resource for all orders matching that customer ID.
 
-
-
-::: bad  
-![Figure: Bad Example – REST API does not return order information with a customer query, so a second API call is needed](graphql-bad-example-underfetching.png)  
+::: bad\
```
Comment (JackDevAU) [inline rules/graphql-when-to-use/rule.md:43]: Hey @ozairashfaqueSSW, 
Please remove the trailing slashes `\` (it's a bug with the Netlify CMS editor)

File: rules/graphql-when-to-use/rule.md [hunk 2]
```diff
@@ -38,61 +36,53 @@ GraphQL differs from REST in that REST is concerned with  **resources** and Grap
...
+![Figure: Bad Example – REST API does not return order information with a customer query, so a second API call is needed](graphql-bad-example-underfetching.png)\
 :::
 
 * **No need for over-fetching:** Continuing the above example, you may decide a workaround is to return details of all orders with all customer queries. However, this overwhelms your client with data it doesn’t need most of the time, just to solve a problem in one scenario. This is called over-fetching (and is the specific problem Facebook set out to resolve with GraphQL).
 
-
-
-::: bad  
-![Figure: Bad Example – REST API returns a whole bunch of data the client doesn’t need – killing bandwidth (and, on mobile, battery)](graph-ql-bad-overfetching.png)  
+::: bad\
```
Comment (JackDevAU) [inline rules/graphql-when-to-use/rule.md:49]: See comment above. Just need to remove the back slash

File: rules/graphql-when-to-use/rule.md [hunk 3]
```diff
@@ -38,61 +36,53 @@ GraphQL differs from REST in that REST is concerned with  **resources** and Grap
...
+![Figure: Bad Example – REST API returns a whole bunch of data the client doesn’t need – killing bandwidth (and, on mobile, battery)](graph-ql-bad-overfetching.png)\
 :::
 
 * **Client Defined Queries:** In a REST API, the CRUD operations are defined by the API, and the client application is constrained by the operations available. This means that any changes required by your UI necessitate changes to your back end as well. With GraphQL you can change your client queries to meet changing UI needs without needing to update your back end.
 
-::: good  
-![Figure: Good Example – Clients can write queries and specify which fields they want the query to return](graphql-good-example-shaped-query.png)  
+::: good\
```
Comment (JackDevAU) [inline rules/graphql-when-to-use/rule.md:55]: same here

File: rules/graphql-when-to-use/rule.md [hunk 4]
```diff
@@ -38,61 +36,53 @@ GraphQL differs from REST in that REST is concerned with  **resources** and Grap
...
 
 * **No caching:** With the exception of the POST HTTP verb, REST calls are idempotent. This means that for any REST call, if the parameters are the same, the result will always be the same. And as REST is built on top of HTTP, and most HTTP stacks have caching built-in, you get client-side caching for free with REST. With GraphQL, this is not possible. If you want to take advantage of caching, it needs to be server-side and requires effort on the part of the developer.
 * **Not suitable for rich content:** GraphQL is a query language, and is fundamentally meant for data. Rich content, like images and videos, is not best suited to GraphQL. GraphQL is a text-based specification, and while it is possible to (for example) encode images into Base64 strings and send them as a field in your GraphQL type, this is not the best way to do it. This is analogous to storing BLOBs in your SQL database – you can do it, but it’s not a good idea.
 
-
-
-::: bad  
-![Figure: Bad Example – Image encoded as Base64 string uses more bandwidth and processing power than necessary, resulting in poor UX](graphql-image-base64.png)  
+::: bad\
```
Comment (JackDevAU) [inline rules/graphql-when-to-use/rule.md:66]: and here

File: rules/graphql-when-to-use/rule.md [hunk 5]
```diff
@@ -38,61 +36,53 @@ GraphQL differs from REST in that REST is concerned with  **resources** and Grap
...
 * **A public database -** If your product *is* your data, and you make it available for people to consume in their own applications, providing a GraphQL endpoint lets them consume the data in a way that meets the needs of their applicaiton, rather than in a way that the API has defined.
 
-
 GraphQL is not a replacement for REST; in fact, it’s often best to use them together. While GraphQL supports changing state on the server through migrations, the true power of GraphQL is in queries. Many create, update, and delete operations are relatively unchanging and can benefit from being maintained as REST resources – think sign-up forms for example. REST is also better for transferring rich content, like images and videos.
 
-
-::: good  
-![Good Example – GraphQL used to construct a complex query with embedded entities, and images are referenced with a URI that the client can use to retrieve from a REST endpoint](good-example-image-url.png)  
+::: good\
```
Comment (JackDevAU) [inline rules/graphql-when-to-use/rule.md:84]: and the last one 😁

**Review by JackDevAU — CHANGES_REQUESTED** [review]

**Review by ozairashfaqueSSW — COMMENTED** [review]
Hi Jack, Can you please check it now.

File: rules/graphql-when-to-use/rule.md [hunk 6]
```diff
@@ -38,61 +36,53 @@ GraphQL differs from REST in that REST is concerned with  **resources** and Grap
...
 
-
 * **No under-fetching:** Sometimes a REST resource may only give you part of what you need. For example, if you need to know all the orders placed by a customer, what date they were ordered, and their current status, but the initial customer search resource only returns the details of the customer, this is called under-fetching. You then need to use the customer’s ID to query your orders resource for all orders matching that customer ID.
 
-
-
-::: bad  
-![Figure: Bad Example – REST API does not return order information with a customer query, so a second API call is needed](graphql-bad-example-underfetching.png)  
+::: bad
+![Figure: Bad Example – REST API does not return order information with a customer query, so a second API call is needed](graphql-bad-example-underfetching.png)\
```
Comment (JackDevAU) [inline rules/graphql-when-to-use/rule.md:44]: Last thing😅Please remove the trailing back slashes at the end of the figures.

File: rules/graphql-when-to-use/rule.md [hunk 7]
```diff
@@ -38,61 +36,53 @@ GraphQL differs from REST in that REST is concerned with  **resources** and Grap
...
 :::
 
 * **No need for over-fetching:** Continuing the above example, you may decide a workaround is to return details of all orders with all customer queries. However, this overwhelms your client with data it doesn’t need most of the time, just to solve a problem in one scenario. This is called over-fetching (and is the specific problem Facebook set out to resolve with GraphQL).
 
-
-
-::: bad  
-![Figure: Bad Example – REST API returns a whole bunch of data the client doesn’t need – killing bandwidth (and, on mobile, battery)](graph-ql-bad-overfetching.png)  
+::: bad
+![Figure: Bad Example – REST API returns a whole bunch of data the client doesn’t need – killing bandwidth (and, on mobile, battery)](graph-ql-bad-overfetching.png)\
```
Comment (JackDevAU) [inline rules/graphql-when-to-use/rule.md:50]: and this one

File: rules/graphql-when-to-use/rule.md [hunk 8]
```diff
@@ -38,61 +36,53 @@ GraphQL differs from REST in that REST is concerned with  **resources** and Grap
...
 :::
 
 * **Client Defined Queries:** In a REST API, the CRUD operations are defined by the API, and the client application is constrained by the operations available. This means that any changes required by your UI necessitate changes to your back end as well. With GraphQL you can change your client queries to meet changing UI needs without needing to update your back end.
 
-::: good  
-![Figure: Good Example – Clients can write queries and specify which fields they want the query to return](graphql-good-example-shaped-query.png)  
+::: good
+![Figure: Good Example – Clients can write queries and specify which fields they want the query to return](graphql-good-example-shaped-query.png)\
```
Comment (JackDevAU) [inline rules/graphql-when-to-use/rule.md:56]: and this one

File: rules/graphql-when-to-use/rule.md [hunk 9]
```diff
@@ -38,61 +36,53 @@ GraphQL differs from REST in that REST is concerned with  **resources** and Grap
...
 * **Not suitable for rich content:** GraphQL is a query language, and is fundamentally meant for data. Rich content, like images and videos, is not best suited to GraphQL. GraphQL is a text-based specification, and while it is possible to (for example) encode images into Base64 strings and send them as a field in your GraphQL type, this is not the best way to do it. This is analogous to storing BLOBs in your SQL database – you can do it, but it’s not a good idea.
 
-
-
-::: bad  
-![Figure: Bad Example – Image encoded as Base64 string uses more bandwidth and processing power than necessary, resulting in poor UX](graphql-image-base64.png)  
+::: bad
+![Figure: Bad Example – Image encoded as Base64 string uses more bandwidth and processing power than necessary, resulting in poor UX](graphql-image-base64.png)\
```
Comment (JackDevAU) [inline rules/graphql-when-to-use/rule.md:67]: ^^

File: rules/graphql-when-to-use/rule.md [hunk 10]
```diff
@@ -38,61 +36,53 @@ GraphQL differs from REST in that REST is concerned with  **resources** and Grap
...
 * **A public database -** If your product *is* your data, and you make it available for people to consume in their own applications, providing a GraphQL endpoint lets them consume the data in a way that meets the needs of their applicaiton, rather than in a way that the API has defined.
 
-
 GraphQL is not a replacement for REST; in fact, it’s often best to use them together. While GraphQL supports changing state on the server through migrations, the true power of GraphQL is in queries. Many create, update, and delete operations are relatively unchanging and can benefit from being maintained as REST resources – think sign-up forms for example. REST is also better for transferring rich content, like images and videos.
 
-
-::: good  
-![Good Example – GraphQL used to construct a complex query with embedded entities, and images are referenced with a URI that the client can use to retrieve from a REST endpoint](good-example-image-url.png)  
+::: good
+![Good Example – GraphQL used to construct a complex query with embedded entities, and images are referenced with a URI that the client can use to retrieve from a REST endpoint](good-example-image-url.png)\
```
Comment (JackDevAU) [inline rules/graphql-when-to-use/rule.md:85]: Last one 😂

**Review by JackDevAU — CHANGES_REQUESTED** [review]
Just spotted some more back slashes🙂

**Review by JackDevAU — APPROVED** [review]

**Pull request merged**